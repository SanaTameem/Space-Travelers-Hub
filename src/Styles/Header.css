* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid black;
  padding: 15px 10px;
  width: 90%;
  margin-bottom: 20px;
}

.logoTitle {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 30%;
  gap: 10px;
}

.linkContainer {
  display: flex;
  width: 31%;
}

.link {
  text-decoration: none;
  color: #0290ff;
  padding: 0 10px;
}

.border {
  border-left: 1px solid #000;
}

.navImg {
  width: 75px;
}

.active {
  text-decoration: underline;
  text-decoration-color: #0290ff;
}
