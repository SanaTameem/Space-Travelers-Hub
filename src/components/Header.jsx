import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../Assets/planet (1).png';
import '../Styles/Header.css';

const Header = () => (
  <>
    <nav className="header">
      <div className="logoTitle">
        <img src={logo} alt="planet" className="navImg" />
        <h1 className="navHeading">Space Travelers&apos; Hub</h1>
      </div>
      <div className="linkContainer">
        <NavLink className="link" to="/" activeclassname="active">Rockets</NavLink>
        <NavLink className="link" to="/missions" activeclassname="active">Missions</NavLink>
        <NavLink className="link" to="/dragons" activeclassname="active">Dragons</NavLink>
        <NavLink className="link border" to="/profile" activeclassname="active">My Profile</NavLink>
      </div>
    </nav>
  </>
);
export default Header;
