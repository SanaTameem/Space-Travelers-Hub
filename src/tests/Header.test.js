import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../components/Header';

describe('Header', () => {
  test('Heading routes are working correclty', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const rocketRout = screen.getByText('Rockets');
    expect(rocketRout).toMatchSnapshot();
  });

  test('Heading routes are working correclty', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const rocketRout = screen.getByText('My Profile');
    expect(rocketRout).toMatchSnapshot();
  });
});
