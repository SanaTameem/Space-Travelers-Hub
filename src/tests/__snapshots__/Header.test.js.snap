// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Heading routes are working correclty 1`] = `
<a
  activeclassname="active"
  aria-current="page"
  class="link active"
  href="/"
>
  Rockets
</a>
`;

exports[`Header Heading routes are working correclty 2`] = `
<a
  activeclassname="active"
  class="link border"
  href="/profile"
>
  My Profile
</a>
`;
